extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -293,7 +295,14 @@ public void registerField(FieldInfo fieldInfo) {
                 case JAVA_CLASS:
                     LOGGER.debugf("Add %s class: %s", (resource.isApplicationClass() ? "APP" : "FWK"),
                             resource.getFullyQualifiedName());
-                    generatedClass.produce(new GeneratedClassBuildItem(resource.isApplicationClass(), resource.getName(),
+                    boolean appClass = resource.isApplicationClass();
```

Comment:
Hm, I think that we should rather try to enhance the application class predicate used internally by ArC: https://github.com/quarkusio/quarkus/blob/master/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java#L113-L119

This way we wouldn't have to modify the `Resource` class etc. WDYT?

Reply:
We will have a circular dependency issue if we use a transformer to remove final methods

Reply:
I see. In that case, we should add a comment here so that we're not trying to reimplement it later ;-).